app/models/daos/OpenIDInfoDAO.scala

```diff
@@ -0,0 +1,117 @@
+package models.daos
+
+import com.mohiva.play.silhouette.api.LoginInfo
+import com.mohiva.play.silhouette.impl.daos.DelegableAuthInfoDAO
+import com.mohiva.play.silhouette.impl.providers.OpenIDInfo
+import javax.inject.Inject
+import play.api.libs.concurrent.Execution.Implicits._
+import play.api.db.slick.DatabaseConfigProvider
+import scala.concurrent.Future
+
+/**
+ * The DAO to store the OpenID information.
+ */
+class OpenIDInfoDAO @Inject() (protected val dbConfigProvider: DatabaseConfigProvider)
+    extends DelegableAuthInfoDAO[OpenIDInfo] with DAOSlick {
+
+  import driver.api._
+
+  protected def openIDInfoQuery(loginInfo: LoginInfo) = for {
+    dbLoginInfo <- loginInfoQuery(loginInfo)
+    dbOpenIDInfo <- slickOpenIDInfos if dbOpenIDInfo.loginInfoId === dbLoginInfo.id
+  } yield dbOpenIDInfo
+
+  protected def addAction(loginInfo: LoginInfo, authInfo: OpenIDInfo) =
+    loginInfoQuery(loginInfo).result.head.flatMap { dbLoginInfo =>
+      DBIO.seq(
+        slickOpenIDInfos += DBOpenIDInfo(authInfo.id, dbLoginInfo.id.get),
+        slickOpenIDAttributes ++= authInfo.attributes.map {
+          case (key, value) => DBOpenIDAttribute(authInfo.id, key, value)
+        })
+    }.transactionally
+
+  protected def updateAction(loginInfo: LoginInfo, authInfo: OpenIDInfo) =
+    openIDInfoQuery(loginInfo).result.head.flatMap { dbOpenIDInfo =>
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2281812354/issues/source?bid=3144114&fileBranchId=3144125#l34)